lib/ash_authentication/token_resource/verifier.ex

```diff
@@ -3,40 +3,69 @@ defmodule AshAuthentication.TokenResource.Verifier do
   The token resource verifier.
   """
 
-  use Spark.Dsl.Transformer
+  use Spark.Dsl.Verifier
   require Ash.Expr
-  alias Spark.{Dsl.Transformer, Error.DslError}
+  require Logger
+  alias Spark.{Dsl.Verifier, Error.DslError}
   import AshAuthentication.Utils
+  import AshAuthentication.Validations.Action
 
   @doc false
   @impl true
-  @spec after?(any) :: boolean()
-  def after?(_), do: true
+  @spec verify(map) :: :ok | {:error, term}
+  def verify(dsl_state) do
+    with :ok <- validate_domain_presence(dsl_state) do
+      maybe_validate_is_revoked_action_arguments(dsl_state)
+    end
+  end
 
-  @doc false
-  @impl true
-  @spec before?(any) :: boolean
-  def before?(_), do: false
+  defp maybe_validate_is_revoked_action_arguments(dsl_state) do
+    case Verifier.get_option(dsl_state, [:token, :revocation], :is_revoked_action_name, :revoked?) do
+      nil ->
+        :ok
 
-  @doc false
-  @impl true
-  @spec after_compile? :: boolean
-  def after_compile?, do: true
+      action_name ->
+        case validate_action_exists(dsl_state, action_name) do
+          {:ok, action} -> validate_is_revoked_action(dsl_state, action)
+          {:error, _} -> :ok
+        end
+    end
+  end
 
-  @doc false
-  @impl true
-  @spec transform(map) ::
-          :ok | {:ok, map} | {:error, term} | {:warn, map, String.t() | [String.t()]} | :halt
-  def transform(dsl_state) do
-    validate_domain_presence(dsl_state)
+  defp validate_is_revoked_action(dsl_state, action) do
+    with :ok <- validate_action_argument_option(action, :jti, :allow_nil?, [true]),
+         :ok <- validate_action_argument_option(action, :token, :allow_nil?, [true]),
+         :ok <- validate_action_option(action, :returns, [:boolean, Ash.Type.Boolean]) do
+      :ok
+    else
+      {:error, _} ->
+        Logger.warning("""
+        Warning while compiling #{inspect(Verifier.get_persisted(dsl_state, :module))}:
+
+        The `:jti` and `:token` options to the `#{inspect(action.name)}` action must allow nil values and it must return a `:boolean`.
+
+        This was an error in our igniter installer previous to version 4.4.9, which allowed revoked tokens to be reused.
+
+        To fix this, run the following command in your shell:
+
+            mix ash_authentication.upgrade 4.4.8 4.4.9
+
+        Or:
+
+          - remove `allow_nil?: false` from these action arguments, and
+          - add `returns :boolean` to the action.
```

Comment:
the `returns` is a second arg to the action

Reply:
that makes it so much harder to find the action with igniter :sob:

Reply:
Fun fact, I guess it can actually be *both* the second arg *or* `returns :boolean` 😅

Reply:
It shouldn't make it much harder to find though, because the function finding tools accept a list of possible rarities.